src/promptflow-core/promptflow/executor/_flow_nodes_scheduler.py

```diff
@@ -23,6 +27,32 @@
 DEFAULT_CONCURRENCY_FLOW = 16
 
 
+def exit_process_with_delay():
+
+    # Add delay to wait for the main thread to exit to guarantee we can export span data.
+    time.sleep(3)
+    # Use os._exit instead of sys.exit, so that the process can stop without
+    # waiting for the thread created by ThreadPoolExecutor to finish.
+    # sys.exit: https://docs.python.org/3/library/sys.html#sys.exit
+    # Raise a SystemExit exception, signaling an intention to exit the interpreter.
+    # Specifically, it does not exit non-daemon thread
+    # os._exit https://docs.python.org/3/library/os.html#os._exit
+    # Exit the process with status n, without calling cleanup handlers, flushing stdio buffers, etc.
+    # Specifically, it stops process without waiting for non-daemon thread.
+    os._exit(0)
+
+
+def signal_handler(sig, frame):
+    """
+    We have no way to cancel running tasks in ThreadPoolExecutor, so we raise KeyboardInterrupt to exit the process.
```

Comment:
Add more detail about why
1 we have a separate thread here
2 raise an exception here

And explain what if we do not do this?

Reply:
Fixed~